pkg/operator/ceph/csi/operator.go

```diff
@@ -0,0 +1,155 @@
+
+	csiCephCluster := &csiopv1a1.CephCluster{}
+	csiCephCluster.Name = cluster.Name
+	csiCephCluster.Namespace = cluster.Namespace
+	if cluster.Spec.CSI.ReadAffinity.Enabled {
+		csiCephCluster.Spec.ReadAffinity.CrushLocationLabels = cluster.Spec.CSI.ReadAffinity.CrushLocationLabels
+	}
+
+	mons := sets.New[string]()
+	for _, mon := range csiClusterConfig.Monitors {
+		mons.Insert(mon.Endpoint)
+	}
+	csiCephCluster.Spec.Monitors = sets.List(mons)
+
+	err := r.client.Create(r.opManagerContext, csiCephCluster)
+	if err != nil {
+		if kerrors.IsAlreadyExists(err) {
+			err = r.client.Update(r.opManagerContext, csiCephCluster)
+			if err != nil {
+				return errors.Wrapf(err, "failed to update CSI-operator ceph cluster CR %s", csiCephCluster.Name)
+			}
+		}
+		return errors.Wrapf(err, "failed to create CSI-operator ceph cluster CR %s", csiCephCluster.Name)
+	}
+
+	logger.Info("Successfully created ceph-CSI operator ceph cluster CR")
+	return nil
+}
+
+func (r *ReconcileCSI) createCSIOpCephCSIConfCR(cluster cephv1.CephCluster) error {
+
+	cephRadosNamespaceList := &cephv1.CephBlockPoolRadosNamespaceList{}
+	err := r.client.List(r.opManagerContext, cephRadosNamespaceList, &client.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(cephRadosNamespaceList.Items) == 0 {
+		logger.Debug("no cephRadosNamespace is found")
+	} else {
+		for _, radosns := range cephRadosNamespaceList.Items {
+			if !radosns.DeletionTimestamp.IsZero() {
+				logger.Debugf("ceph radosnamespace %q is being deleting", cluster.Namespace)
+				return nil
+			}
+
+			csiOpConfig := &csiopv1a1.Config{}
+			csiOpConfig.Name = csiOPconfigCRName(radosns.Name)
+			csiOpConfig.Spec = csiopv1a1.ConfigSpec{
+				CephClusterRef: v1.LocalObjectReference{
+					Name: cluster.Name,
+				},
+				Rbd: &csiopv1a1.RbdConfigSpec{
+					RadosNamespace: radosns.Name,
+				},
+			}
+
+			err = r.client.Create(r.opManagerContext, csiOpConfig)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI-operator config cr for RBD %s", csiOpConfig.Name)
+			}
+		}
+	}
+
+	cephSubvolumeGroupList := &cephv1.CephFilesystemSubVolumeGroupList{}
+	err = r.client.List(r.opManagerContext, cephSubvolumeGroupList, &client.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(cephSubvolumeGroupList.Items) == 0 {
+		logger.Debug("no ceph subvolumegroup found")
+	} else {
+		for _, cephsvg := range cephSubvolumeGroupList.Items {
+			if !cephsvg.DeletionTimestamp.IsZero() {
+				logger.Debugf("ceph radosnamespace %q is being deleting", cluster.Namespace)
+				return nil
+			}
+
+			csiOpConfig := &csiopv1a1.Config{}
+			csiOpConfig.Name = csiOPconfigCRName(cephsvg.Name)
+			csiOpConfig.Spec = csiopv1a1.ConfigSpec{
+				CephClusterRef: v1.LocalObjectReference{
+					Name: cluster.Name,
+				},
+				CephFs: &csiopv1a1.CephFsConfigSpec{
+					SubVolumeGroup: cephsvg.Name,
+				},
+			}
+
+			err = r.client.Create(r.opManagerContext, csiOpConfig)
```

Comment:
Similarly, seems like this should be created/updated with the subvolumegroup controller.